pkg/build/controller/build/build_controller.go

```diff
@@ -364,14 +375,24 @@ func (bc *BuildController) createPodSpec(originalBuild *buildapi.Build, ref stri
 	return podSpec, nil
 }
 
-// resolveOutputDockerImageReference returns a reference to a Docker image
-// computed from the buid.Spec.Output.To reference.
-func (bc *BuildController) resolveOutputDockerImageReference(build *buildapi.Build) (string, error) {
+// resolveOutputDockerImageReference returns a Docker image name
+// computed from the build.Spec.Output.To reference.  It also attempts to find a Secret
+// associated with the build service account which can push to the image location and
+// return that, if found.
+// Note that we are using controller level permissions to resolve the imagestream and secret,
+// meaning users could theoretically define a build that references an imagestream they cannot
+// see, and 1) get the docker image reference of that imagestream and 2) a reference to a secret
+// associated with a service account that can push to that location.  However they still cannot view the secret,
+// and ability to use a service account implies access to its secrets, so this is considered safe.
```

Comment:
This bit seems more convincing.  They're allowed to create builds that are allowed to push to targets, so preventing them from seeing the secret used to push the resultant image seems less useful.

Going the other way (why would I want to restrict this) would be story like this:
 1. I want people to be able to create strictly shaped pods to run builds and produce images, but I do not want them to create general pods.
 2. I want to prevent people from getting the power to push whatever they want. Instead, I will force them to run builds against my source control and thereby have a set of istags that are all relatable to a source code revision.

Would we want this for tagging our openshift images in the docker registry?  David has the power to kick a build, but no matter how clever he is, he does not have the power to push and tag images himself?